execution_engine_testing/test_support/src/wasm_test_builder.rs

```diff
@@ -174,6 +181,35 @@ impl InMemoryWasmTestBuilder {
             ..Default::default()
         }
     }
+
+    /// Returns an [`InMemoryWasmTestBuilder`] instantiated using values from a given chainspec.
+    pub fn new_with_chainspec<P: AsRef<Path>>(
+        chainspec_path: P,
+        post_state_hash: Option<Digest>,
+    ) -> Self {
+        let chainspec_config = ChainspecConfig::from_chainspec_path(chainspec_path)
+            .expect("must build chainspec configuration");
+
+        let engine_config = EngineConfig::new(
+            DEFAULT_MAX_QUERY_DEPTH,
+            chainspec_config.core_config.max_associated_keys,
+            chainspec_config.core_config.max_runtime_call_stack_height,
+            chainspec_config.core_config.minimum_delegation_amount,
+            chainspec_config.core_config.strict_argument_checking,
+            chainspec_config.wasm_config,
+            chainspec_config.system_costs_config,
+        );
+
+        let global_state = InMemoryGlobalState::empty().expect("should create global state");
+
+        Self::new(global_state, engine_config, post_state_hash)
+    }
+
+    /// Returns an [`InMemoryWasmTestBuilder`] instantiated
+    /// using values from the production chainspec.
+    pub fn new_with_production_chainspec() -> Self {
```

Comment:
I agree with @goral09 point that this method should be removed and implemented as `Default` instead.

Reply:
Removed, and implemented as `Default`